logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/ephemeral/EphemeralMessageDeletionHandlerTest.kt

```diff
@@ -0,0 +1,413 @@
+                )
+            )
+        ).withMessageRepositoryMarkingSelfDeletionStartDate().arrange()
+
+        // when
+        ephemeralMessageDeletionHandler.startSelfDeletion(
+            conversationId = ConversationId("someValue", "someDomain"),
+            messageId = "someId"
+        )
+
+        advanceUntilIdle()
+
+        // then
+        verify(arrangement.messageRepository)
+            .suspendFunction(arrangement.messageRepository::markSelfDeletionStartDate)
+            .with(any(), any(), any())
+            .wasInvoked(exactly = Times(1))
+
+        verify(arrangement.messageRepository)
+            .suspendFunction(arrangement.messageRepository::getMessageById)
+            .with(any(), any())
+            .wasInvoked(exactly = Times(1))
+    }
+
+    @Test
+    fun givenRegularMessage_whenEnqueueingTwice_thenSelfDeletionShouldBeCalledOnce() = runTest(testDispatcher.default) {
+        // given
+        val (arrangement, ephemeralMessageDeletionHandler) = Arrangement(this, testDispatcher)
+            .withMessageRepositoryReturningMessage(
+                message = TestMessage.TEXT_MESSAGE.copy(
+                    expirationData = Message.ExpirationData(
+                        expireAfter = 1.seconds,
+                        selfDeletionStatus = Message.ExpirationData.SelfDeletionStatus.NotStarted
+                    )
+                )
+            )
+            .withMessageRepositoryMarkingSelfDeletionStartDate()
+            .arrange()
+
+        // when
+        ephemeralMessageDeletionHandler.startSelfDeletion(
+            conversationId = ConversationId("someValue", "someDomain"),
+            messageId = "someId"
+        )
+
+        ephemeralMessageDeletionHandler.startSelfDeletion(
+            conversationId = ConversationId("someValue", "someDomain"),
+            messageId = "someId"
+        )
+
+        advanceUntilIdle()
+
+        // then
+        verify(arrangement.messageRepository)
+            .suspendFunction(arrangement.messageRepository::markSelfDeletionStartDate)
+            .with(any(), any(), any())
+            .wasInvoked(exactly = Times(1))
+
+        verify(arrangement.messageRepository)
+            .suspendFunction(arrangement.messageRepository::getMessageById)
+            .with(any(), any())
+            .wasInvoked(exactly = Times(2))
+    }
+
+    @Test
+    fun givenRegularMessageWithExpiration_whenEnqueueForDeletionAndTimeElapsed_thenTheMessageShouldBeDeleted() =
+        runTest(testDispatcher.default) {
+            // given
+            val timeUntilExpiration = 1.seconds
+
+            val (arrangement, ephemeralMessageDeletionHandler) = Arrangement(
+                coroutineScope = this,
+                dispatcher = testDispatcher
+            ).withMessageRepositoryReturningMessage(
+                message = TestMessage.TEXT_MESSAGE.copy(
+                    expirationData = Message.ExpirationData(
+                        expireAfter = 1.seconds,
```

Comment:
Shouldn't you pass `timeUntilExpiration` here instead of defining the absolute value?